File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        s1 = 0
        for t1 in t:
            if len(s) == 0:
                return True
            if s[s1] == t1:
                s1 += 1
            if s1 >= len(s):
                break
                return False
        if s1 == len(s):
            return True
        else:
            return False

File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if(s == ""){
            return true;
        }
        bool flag = false;
        int j =0;
        for(int i =0; i<t.length(); i++){
            if(s[j] == t[i]){
                j++;
                if(j == s.length()){
                    flag = true;
                    break;
                }
            }
        }
        return flag;
        
    }
};